streaming/src/main/scala/org/apache/spark/streaming/api/java/JavaStreamingListener.scala

```diff
@@ -0,0 +1,189 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.api.java
+
+import org.apache.spark.annotation.DeveloperApi
+import org.apache.spark.streaming.Time
+import org.apache.spark.streaming.scheduler._
+
+/**
+ * A listener interface for receiving information about an ongoing streaming  computation.
+ */
+@DeveloperApi
+abstract class JavaStreamingListener {
+
+  /** Called when a receiver has been started */
+  def onReceiverStarted(receiverStarted: JavaStreamingListenerReceiverStarted): Unit = { }
+
+  /** Called when a receiver has reported an error */
+  def onReceiverError(receiverError: JavaStreamingListenerReceiverError): Unit = { }
+
+  /** Called when a receiver has been stopped */
+  def onReceiverStopped(receiverStopped: JavaStreamingListenerReceiverStopped): Unit = { }
+
+  /** Called when a batch of jobs has been submitted for processing. */
+  def onBatchSubmitted(batchSubmitted: JavaStreamingListenerBatchSubmitted): Unit = { }
+
+  /** Called when processing of a batch of jobs has started.  */
+  def onBatchStarted(batchStarted: JavaStreamingListenerBatchStarted): Unit = { }
+
+  /** Called when processing of a batch of jobs has completed. */
+  def onBatchCompleted(batchCompleted: JavaStreamingListenerBatchCompleted): Unit = { }
+
+  /** Called when processing of a job of a batch has started. */
+  def onOutputOperationStarted(
+      outputOperationStarted: JavaStreamingListenerOutputOperationStarted): Unit = { }
+
+  /** Called when processing of a job of a batch has completed. */
+  def onOutputOperationCompleted(
+      outputOperationCompleted: JavaStreamingListenerOutputOperationCompleted): Unit = { }
+}
+
+/**
+ * :: DeveloperApi ::
+ * Base trait for events related to JavaStreamingListener
+ */
+@DeveloperApi
+sealed trait JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerBatchSubmitted(val batchInfo: JavaBatchInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerBatchCompleted(val batchInfo: JavaBatchInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerBatchStarted(val batchInfo: JavaBatchInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerOutputOperationStarted(val outputOperationInfo: JavaOutputOperationInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerOutputOperationCompleted(
+    val outputOperationInfo: JavaOutputOperationInfo) extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerReceiverStarted(val receiverInfo: JavaReceiverInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerReceiverError(val receiverInfo: JavaReceiverInfo)
+  extends JavaStreamingListenerEvent
+
+@DeveloperApi
+class JavaStreamingListenerReceiverStopped(val receiverInfo: JavaReceiverInfo)
```

Comment:
The JavaSparkListener uses the same object types as SparkListener in the method. You seem to have defined additional Java classes, why? To avoid options?